src/5.3/libu/bread.p

```diff
@@ -0,0 +1,220 @@
+            and (str[2] = 'i')
+            and (str[3] = 'n')
+            and (str[4] = 'f'))
+        then begin
+            str[2] := '9';
+            str[3] := '9';
+            str[4] := '.';
+            str[5] := '9';
+            str[6] := 'e';
+            str[7] := '9';
+            str[8] := '9';
+            str[9] := '9';
+            len := 9;
+        end;
+    end;
+
+begin
+    if (ugeteof() = 1) then begin
+        u.Opc := Ueof;
+    end else begin
+        u.Intarray[1] := ugetint();
+        u.Intarray[2] := ugetint();
+        urec := utab[u.Opc]; 
+        instlength := urec.instlength;
+        
+        i := 3;
+        while i <> instlength + 1 do begin
+            u.Intarray[i] := ugetint();
+            u.Intarray[i + 1] := ugetint();
+            i := i + 2;
+        end;
+
+        if (urec.hasconst) then begin
+            u.Intarray[instlength + 1] := ugetint();
+            u.Intarray[instlength + 2] := ugetint();
+
+            if ((u.Dtype in [Mdt, Qdt, Rdt, Sdt, Xdt]) or (u.Opc = Ucomm)) then begin
+            
+                strlength := (u.Intarray[instlength + 1] + 3) div 4;
+
+                if ((strlength & 1) <> 0) then begin
+                    strlength := strlength + 1;
+                end;
+
+                i := 1;
+                while i <> strlength + 1 do begin
+                    strptr^.ssarray[i] := ugetint();
+                    strptr^.ssarray[i + 1] := ugetint();
+                    i := i + 2;
+                end;
+
+                if ((u.Dtype in [Qdt, Rdt])) then begin
+                    if (u.Opc <> Uinit) then begin
+                        fix_infinity(strptr^.ss, u.Constval.Ival);
+                    end else begin
+                        fix_infinity(strptr^.ss, u.Initval.Ival);
+                    end;
+                end;
+
+                if (u.Opc <> Uinit) then begin
+                    u.Constval.Chars := strptr;
+                end else begin
+                    u.Initval.Chars := strptr;
+                end;
+            end;
+        end;
+    end;
+end;
+
+procedure initur(arg0: Filename);
+begin
+    ugetinit(arg0);
+    datachars := ['A', 'C', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'W', 'X', 'Z'];    
+    memorychars := ['A', 'M', 'P', 'R', 'S', 'Z'];
+    setconstantchars := ['0'..'9', 'A'..'F'];
+    
+    dtytype['A'] := Adt;
+    dtytype['C'] := Cdt;
+    dtytype['F'] := Fdt;
+    dtytype['G'] := Gdt;
+    dtytype['H'] := Hdt;
+    dtytype['I'] := Idt;
+    dtytype['J'] := Jdt;
+    dtytype['K'] := Kdt;
+    dtytype['L'] := Ldt;
+    dtytype['M'] := Mdt;
+    dtytype['N'] := Ndt;
+    dtytype['P'] := Pdt;
+    dtytype['Q'] := Qdt;
+    dtytype['R'] := Rdt;
+    dtytype['S'] := Sdt;
+    dtytype['W'] := Wdt;
+    dtytype['X'] := Xdt;
+
+    mtytype['Z'] := Zmt;
+    mtytype['M'] := Mmt;
+    mtytype['P'] := Pmt;
+    mtytype['R'] := Rmt;
+    mtytype['S'] := Smt;
+    mtytype['A'] := Amt;
+end;
+
+{UNUSED}
```

Comment:
since this is library code, any function may be used or not used in different executables, so this comment may be misleading